modules/apps/portal-configuration/portal-configuration-upgrade-impl/src/main/java/com/liferay/portal/configuration/upgrade/internal/PrefsPropsToConfigurationUpgradeHelperImpl.java

```diff
@@ -136,10 +165,16 @@ else if (returnType == String[].class) {
 			return;
 		}
 
-		Configuration configuration = _configurationAdmin.getConfiguration(
-			configurationClass.getName(), StringPool.QUESTION);
+		if (companyId > 0) {
+			_configurationProvider.saveCompanyConfiguration(
+				configurationClass, companyId, properties);
+		}
+		else {
+			Configuration configuration = _configurationAdmin.getConfiguration(
+				configurationClass.getName(), StringPool.QUESTION);
 
-		configuration.update(properties);
+			configuration.update(properties);
```

Comment:
For consistency, we can use `ConfigurationProvider.saveSystemConfiguration` here for the same result. The only difference is that `ConfigurationProvider` will [get the pid from the configurationClass Metatype](https://github.com/liferay/liferay-portal/blob/31459ed4ca6f158bf206b493da30c6c303070d38/modules/apps/portal-configuration/portal-configuration-module-configuration/src/main/java/com/liferay/portal/configuration/module/configuration/internal/ConfigurationProviderImpl.java#L263) instead of the class name. In practice they should always be the same.